### PR TITLE
add etcd-grpc

### PR DESCRIPTION
**Pull request by borune-k12**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/etcd-grpc/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    homepage = "https://github.com/trassir/etcd-grpc"
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+	"shared": True,
```
Comment (Artalus) [inline recipes/etcd-grpc/all/conanfile.py]: выравнивание съехало

File: recipes/etcd-grpc/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+class EtcdGrpcConan(ConanFile):
+    name = "etcd-grpc"
+    description = "DSSL etcd gRPC headers library"
+    topics = ("conan", "grpc", "protobuf", "etcd")
+    url = "https://github.com/trassir/etcd-grpc"
+    homepage = "https://github.com/trassir/etcd-grpc"
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler"
```
Comment (Artalus) [inline recipes/etcd-grpc/all/conanfile.py]: `build_type`?

File: recipes/etcd-grpc/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        if self.options.shared:
+            cmake.definitions["CONAN_BUILD_SHARED_LIBS"] = "ON"
+
+        cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        protoc_module = self.deps_cpp_info["protoc"]
+        protoc_path = os.path.join(protoc_module.rootpath, protoc_module.bindirs[0], "protoc")
```
Comment (Artalus) [inline recipes/etcd-grpc/all/conanfile.py]: а ту же переменную окружения `PROTOC_BIN` здесь не получится подцепить?

Comment (borune-k12) [reply recipes/etcd-grpc/all/conanfile.py]: неа

**Pull request merged**